src/Config/FileSystemRuntimeConfigLoader.cs

```diff
@@ -154,7 +160,10 @@ public string GetFileNameForEnvironment(string? aspnetEnvironment, bool consider
     /// <returns></returns>
     public string GetFileName(string? environmentValue, bool considerOverrides)
     {
-        string fileNameWithoutExtension = _fileSystem.Path.GetFileNameWithoutExtension(_baseConfigFileName);
+        // If the baseConfigFileName contains a path, we need to ensure that it is not lost. for example: baseConfigFileName = "config/dab-config.json"
+        // in this case, we need to get the directory name and the file name without extension and then combine them back. Else, we will lose the path
+        // and the file will be searched in the current directory.
+        string fileNameWithoutExtension = _fileSystem.Path.Combine(_fileSystem.Path.GetDirectoryName(_baseConfigFileName) ?? string.Empty, _fileSystem.Path.GetFileNameWithoutExtension(_baseConfigFileName));
```

Comment:
You have code here which preserves the path (per comment), however, the next few lines of code check `DoesFileExistInCurrentDirectory()` which would presumably fail and hit line 185 to return empty string.

Reply:
It won't fail because the file name would contain the path details as well. for example if fileName is `/dab/configs/dab-config.json` and we are in root directory(`/`), then it would search the file using the directory information contained in the file Name. which is why it was failing before because earlier it used to search for `dab-config.json` in the current directory as the directory info was getting lost.

Reply:
updated PR description as well